### PR TITLE
Documentation and error message enhancements for GraphOfTheGodsFactory with no index backend configured

### PR DESCRIPTION
**Pull request by cjquinon**
This changeset updates the getting-started documentation to indicate that a user must use the `loadWithoutMixedIndex` method when using a JanusGraph configuration without an indexing backend.  It also updates the "load" method to throw an exception when it's told to use mixed indexes, but an indexing backend isn't configured.

Issue: #298

### CODE REVIEW

File: janusgraph-core/src/main/java/org/janusgraph/example/GraphOfTheGodsFactory.java [hunk 1]
```diff
@@ -61,6 +62,14 @@ public static void load(final JanusGraph graph) {
     }
 
     public static void load(final JanusGraph graph, String mixedIndexName, boolean uniqueNameCompositeIndex) {
+        if (mixedIndexName != null
+                && graph instanceof StandardJanusGraph
+                && ((StandardJanusGraph)graph).getIndexSerializer().containsIndex(mixedIndexName) == false) {
+            String msg = "The indexing backend with name \"%s\" is not defined. Specify an existing index " +
```
Comment (amcp) [inline janusgraph-core/src/main/java/org/janusgraph/example/GraphOfTheGodsFactory.java]: externalize the string format to a constant

File: janusgraph-core/src/main/java/org/janusgraph/example/GraphOfTheGodsFactory.java [hunk 2]
```diff
@@ -61,6 +62,14 @@ public static void load(final JanusGraph graph) {
...
+                         "backend or use GraphOfTheGodsFactory.loadWithoutMixedIndex(graph,true) to load " +
+                         "without the use of an indexing backend.";
+            throw new IllegalStateException(String.format(msg, mixedIndexName));
```
Comment (amcp) [inline janusgraph-core/src/main/java/org/janusgraph/example/GraphOfTheGodsFactory.java]: use Preconditions.checkState

File: janusgraph-core/src/main/java/org/janusgraph/example/GraphOfTheGodsFactory.java (hunk 1 above)
Comment (cjquinon) [inline janusgraph-core/src/main/java/org/janusgraph/example/GraphOfTheGodsFactory.java]: Fixed!

File: janusgraph-core/src/main/java/org/janusgraph/example/GraphOfTheGodsFactory.java (hunk 2 above)
Comment (cjquinon) [inline janusgraph-core/src/main/java/org/janusgraph/example/GraphOfTheGodsFactory.java]: Fixed!

**Comment by amcp** [comment]
@cjquinon the build is broken: https://travis-ci.org/JanusGraph/janusgraph/jobs/239120646#L1266
You have not imported Preconditions. please add and start a new build

**Comment by cjquinon** [comment]
@amcp : sorry about that.  Fixed!

**Pull request merged**